hibernate-core/src/main/java/org/hibernate/graph/EntityGraphSemantic.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.graph;
+
+/**
+ * JPA defines 2 distinct semantics for applying an EntityGraph.  This
+ * enumeration captures those 2 semantics (plus a `null` value).
+ *
+ * @author Steve Ebersole
+ */
+public enum EntityGraphSemantic {
+	/**
+	 * Indicates a "fetch graph" EntityGraph.  Attributes explicitly specified
+	 * as AttributeNodes are treated as FetchType.EAGER (via join fetch or
+	 * subsequent select).
+	 * <p/>
+	 * Note: Currently, attributes that are not specified are treated as
+	 * FetchType.LAZY or FetchType.EAGER depending on the attribute's definition
+	 * in metadata, rather than forcing FetchType.LAZY.
+	 */
+	FETCH( "javax.persistence.fetchgraph" ),
+
+	/**
+	 * Indicates a "load graph" EntityGraph.  Attributes explicitly specified
+	 * as AttributeNodes are treated as FetchType.EAGER (via join fetch or
+	 * subsequent select).  Attributes that are not specified are treated as
+	 * FetchType.LAZY or FetchType.EAGER depending on the attribute's definition
+	 * in metadata
+	 */
+	LOAD( "javax.persistence.loadgraph" ),
+
+	/**
+	 * No graph
+	 */
+	NONE( null );
```

Comment:
Not sure how this helps. Isn't the deciding factor the existence of the graph itself. If the graph is not null but this is set to NONE what would that mean?

Reply:
Nit-pick much lol.  How does it hurt?  And I thought it helped.  Is there a real reason to remove it?